src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests_WithDigitSeparators.cs

```diff
@@ -0,0 +1,2137 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeCleanup;
+using Microsoft.CodeAnalysis.CodeCleanup.Providers;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.CodeCleanup
+{
+    public class ReduceTokenTests_WithDigitSeparators
+    {
+        [Fact]
+        [WorkItem(5529, "DevDiv_Projects/Roslyn")]
+        [Trait(Traits.Feature, Traits.Features.ReduceTokens)]
+        [Trait(Traits.Feature, "WithDigitSeparators")]
+        public async Task ReduceSingleLiterals_LessThan8Digits()
```

Comment:
1. Seems very odd that VB specific tests are in the core workspace layer.
2. The work item string seems wrong.  it shoudl link to the github bug.

Reply:
It looks like you expanded the check from integral tokens to decimal and floating point tokens.  It seems like it would suffice to have a couple of tests for both of those types.  in other words, not 2k lines of tests, but more like 20 :)

Reply:
@CyrusNajmabadi 
It surprised me to that the unit test were that for up the layers, in a core (shared) section and in are in C# not VB.

I was planning to duplicate the tests with digit separator, so we can check that the simplified versions are valid. Especially around rounding and overflow. Which is currently an open issue #22781